### ISSUE TITLE
[regression] CFs/metas are ignored without permalinks

### ISSUE THREAD

**Issue by dev-101** [osclass/Osclass#2259]
Filtering by Custom Fields (e.g. tested with simple dropdown CF) does not work anymore without permalinks.

**Comment by dev-101** [comment]
Preliminary testing shows that it is related to these changes here:

https://github.com/osclass/Osclass/commit/2d2d03589fa1ffde84b0935c6c1fdbd2eb83e82d#diff-d6335c50f2d0536514781dc51d34f72a
https://github.com/osclass/Osclass/commit/fdc1535cba99e893103f3c3f4da5227ea7a621e2#diff-d6335c50f2d0536514781dc51d34f72a

https://github.com/osclass/Osclass/commit/0dfcea71a53b2b8b090e5846b53e26dca7fe8d66#diff-d6335c50f2d0536514781dc51d34f72a

**Comment by dev-101** [comment]
After more investigation, bug is definitely related to extractParams() method and parse_str()
It erroneously parses meta[] parameters, for example:

```meta[3]=3``` parameter

becomes ```meta``` with meta key/value pairs stored as another parameter in array - e.g. ```[3=>3]```

**Comment by garciademarina** [comment]
Thanks @dev-101 I will take a look.

**Comment by garciademarina** [comment]
I think I found out the issue... I will need some testing. 

moving ```$this->extractParams($request_uri);``` from line 166, one line up. 
extractParams it's need only if permalinks are enabled, so you  extractParams shoud be inside ```if(osc_rewrite_enabled()) {```

https://github.com/osclass/Osclass/blob/master/oc-includes/osclass/classes/Rewrite.php#L166

Can someone test it a bit ? ( @dev-101 )

**Comment by dev-101** [comment]
Yes, I can confirm that it is working. This was in fact my original thought at first, to restrict it for permalinks only, but because of some other issues, I got focused on extractParams instead. Thanks!
